File: 171.excel表列序号.py
#
# @lc app=leetcode.cn id=171 lang=python
#
# [171] Excel表列序号
#

# @lc code=start
class Solution(object):
    def titleToNumber(self, s):
        """
        :type s: str
        :rtype: int
        """
        res = 0
        bit = 1
        for a in s[::-1]:
            res += (ord(a) - 64) * bit
            bit *= 26
        return res
# @lc code=end

File: 171.excel表列序号.cpp
/*
 * @lc app=leetcode.cn id=171 lang=cpp
 *
 * [171] Excel表列序号
 */

// @lc code=start
class Solution {
public:
    int titleToNumber(string s) {
        reverse(s.begin(), s.end());
        int sum = 0;
        for(int i = 0; i < s.size(); i++){
            sum += (s[i] - 'A' + 1)*pow(26, i);
        }
        return sum;

    }
};
// @lc code=end

